### ISSUE TITLE
Modifying children causes zoom level to reset

### ISSUE THREAD

**Issue by tybro0103** [tomchentw/react-google-maps#68]
This can be seen in the demo page. Zoom in, then click to add a marker. It will zoom back out. Same thing for removing markers.

**Comment by tomchentw** [comment]
Since this issue has been mentioned over and over again, I was thinking if we should make `<GoogleMaps>` a component like `<input>`. IE, they can have a **controlled** state and **uncontrolled** state. For most usage, **uncontrolled** `<GoogleMaps>` is enough. But that requires us to put the config into `props.defaultConfig` or something ...

**Comment by ryanannatuvasquez** [comment]
:+1: @tomchentw  Is there any snippet can you provide or example?

**Comment by khoerling** [comment]
+2

**Comment by wuct** [comment]
:+1: 
@ryanannatuvasquez It has not been implemented, yet. According to [React doc](https://facebook.github.io/react/docs/forms.html), I believe it will be something like below:
```
<GoogleMaps
  defaultZoom={3}
  defaultCenter={{lat: -25.363882, lng: 131.044922}} />
```

**Comment by tomchentw** [comment]
Thanks @wuct ! This is exactly what I'd expect. Please leave any feedback here.
@ryanannatuvasquez @khoerling @tybro0103

**Comment by ryanannatuvasquez** [comment]
@tomchentw I came up with temporary solution, having `zoomLevel` and `location` state, the `zoomLevel` updates in onZoomChanged` event and `onDragend` event for `location`.

**Comment by tomchentw** [comment]
@ryanannatuvasquez That's the standard ("React") way to handle internal state of a component. Good job!

**Comment by tomchentw** [comment]
Please move discussions to #88, thanks. Close this now.

**Comment by tomchentw** [comment]
Released [v2.0.0] and [v2.0.1].

[v2.0.0]: https://github.com/tomchentw/react-google-maps/releases/tag/v2.0.0
[v2.0.1]: https://github.com/tomchentw/react-google-maps/releases/tag/v2.0.1
